base/linalg/generic.jl

```diff
@@ -957,7 +957,7 @@ true
 function istriu(A::AbstractMatrix)
     m, n = size(A)
     for j = 1:min(n,m-1), i = j+1:m
-        if A[i,j] != 0
+        if A[i,j] != zero(eltype(A))
```

Comment:
IIUC, `!iszero(A[i,j])` should be preferred: It may avoid creation of a zero instance (e.g. for `BigInt`) and furthermore, there are types `T` for which `zero(T)` is not defined although `iszero(x::T)` is (e.g. matrices).

Applies again below.